lib/checkassert.cpp

```diff
@@ -120,37 +100,173 @@ void CheckAssert::assignmentInAssertError(const Token *tok, const std::string& v
                 "builds, this is a bug.", CWE398, Certainty::normal);
 }
 
-// checks if side effects happen on the variable prior to tmp
-void CheckAssert::checkVariableAssignment(const Token* assignTok, const Scope *assertionScope)
-{
-    if (!assignTok->isAssignmentOp() && assignTok->tokType() != Token::eIncDecOp)
-        return;
+//---------------------------------------------------------------------------
+
+/**
+ * \brief checks if a variable's value may be changed with the given operator
+ * \retval true   - is variable's value may change
+ * \retval false  - no possible value altering
+ */
+bool CheckAssert::isVariableValueChangingOperator(const Token *token) {
+    if (token->isAssignmentOp())
+        return true;
+    if (token->isIncDecOp())
+        return true;
 
-    const Variable* var = assignTok->astOperand1()->variable();
+    return false;
+}
+
+/**
+ * \brief checks if assignment of variable has side effect
+ * \param var             - variable a value is assigned to
+ * \param assertionScope  - scope of the assertion
+ * \retval true           - the assignment has a possible side effect
+ * \retval false          - no side effect from assignment
+ */
+bool CheckAssert::checkVariableAssignmentSideEffect(const Variable *var, const Scope *assertionScope) {
```

Comment:
I don't like this function name. not sure what to write but something like `is..` or `has..` would be better.
This `check..` sounds to me like it might output some error messages and stuff..